[RFC] vim-patch:8.1.0053 first argument of 'completefunc' has inconsistent type

cangscop opened the pull request:
Problem:    The first argument given to 'completefunc' can be Number or
            String, depending on the value.
Solution:   Avoid guessing the type of an argument, use typval_T in the
            callers of call_vim_function(). (Ozaki Kiichi)
vim/vim#2993

cangscop committed:
vim-patch:8.1.53 use typval_T in the caller of call_vim_function
Problem:	unreliable types for complete function arguments
Solution:	fix argument type for functions w/ unreliable type conversion(Ozaki Kiichi)
vim/vim#2993

justinmk commented:
shoudl be `vim-patch:8.1.0053` (which is a valid Vim git tag) not `vim-patch:8.1.53` (which is not a valid Vim git tag). `scripts/vim-patch.sh` would do this for you next time :)

cangscop committed:
vim-patch:8.1.0053 use typval_T in the caller of call_vim_function
Problem:	unreliable types for complete function arguments
Solution:	fix argument type for functions w/ unreliable type conversion(Ozaki Kiichi)
vim/vim#2993

cangscop committed:
vim-patch:8.1.0053 use typval_T in the caller of call_vim_function
Problem:	unreliable types for complete function arguments
Solution:	fix argument type for functions w/ unreliable type conversion(Ozaki Kiichi)
vim/vim#2993

justinmk commented on src/nvim/eval.c:
```diff
@@ -1275,116 +1275,80 @@ int get_spellword(list_T *const list, const char **ret_word)
-    int str_arg_only,               // all arguments are strings
+    typval_T *argv,
+    int safe,                       // use the sandbox
     typval_T *rettv
```
in the vim patch, `safe` is the last arg. what's the reason for changing the order? 

Also `safe` could stay as `bool`.

justinmk commented on src/nvim/eval.c:
```diff
@@ -1275,116 +1275,80 @@ int get_spellword(list_T *const list, const char **ret_word)
-// Uses argv[argc] for the function arguments.  Only Number and String
-// arguments are currently supported.
+// Uses argv[argc-1] for the function arguments. argv[argc]
+// should have type VAR_UNKNOWN.
```
the comment is different in the vim patch. why change it?

janlazo commented:
vim-patch:8.1.0055 changes were included in this PR so the patch is N/A now.
